### ISSUE TITLE
Live preview could keep its state the same

### ISSUE THREAD

**Issue by lmatteis** [enjalot/blockbuilder#136]
Hi there. So I've talked about this on Twitter but I was thinking that the live-preview feature is awesome but at the same time, you lose the state you were in. I was thinking, since D3 exposes the update sets (from the joins), why not keep the live preview DOM the same, and re-execute the JS on top of it - if it's well written d3 (with well handled enter/update/exit sets), then the preview should update seamlessly.

I would love this feature because I'm often building complex interactions, and I need to change parameters at a specific state I am in (after having clicked a bunch of buttons). Currently the preview is redrawn and I have to repeat my interactions every time.

Something similar could also be done for the stylesheets - figure out if they change and only update the `<style/>` portions in the DOM.

**Comment by curran** [comment]
It might be interesting to experiment with storing the state somewhere persistent, like [local storage](https://developer.mozilla.org/en/docs/Web/API/Window/localStorage).

**Comment by enjalot** [comment]
I replied to your thread on the d3 mailing list, where I think this might be better discussed. In short I experimented with these techniques in http://tributary.io
It would be great to see this kind of thing working in blockbuilder, but it would be difficult to do while maintaining blockbuilder's core design principle (it should work just like a browser + local file). It would be interesting if a fork of blockbuilder went in this direction though!
